### ISSUE TITLE
Convert selicEntries return to a typed return

### ISSUE THREAD

**Issue by adilsoncarvalho** [marileize/selic#6]
Create a base type to carry the Selic index entry that can be used to return the results from the function. The types should use actual types for data, not just strings.

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html

**Comment by marileize** [comment]
@adilsoncarvalho please check if I understood the issue.

I'm planning to create a class `SelicIndexEntry` with the properties:
```
date: Date;
rate: number;
```

Is it?

**Comment by adilsoncarvalho** [comment]
Yes, but think of them as a record, meaning that both (date, rate) are a single unit of data.

**Comment by adilsoncarvalho** [comment]
1. declare a type for the selic tuple
2. create a function to convert date string to a Date object (tests! tests! tests!)
3. return an array of selic tuples
4. tests! tests! tests!

```typescript
[1, 2, 3, 4].map((n) => n.toString()) // ["1", "2", "3", "4"]

function convert({data: string, valor: string}): {date: Date, value: number} {
  return {
    date: new Date(string),
    value: parseFloat(valor),
  }
}

(await response.json()).map(convert)
```

**Comment by adilsoncarvalho** [comment]
Give a look to this documentation about iterators for collections.

https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html

**Comment by adilsoncarvalho** [comment]
https://runkit.com/adilson/counting-names-in-a-list

**Comment by marileize** [comment]
@adilsoncarvalho, regarding "declare a type for the selic tuple", do you expect that?

```
type SelicEntry = {
 rateDate: Date;
 rateValue: number;
};
```
